src/python/dxpy/templating/templates/nextflow/src/nextflow.sh

```diff
@@ -69,18 +69,49 @@ on_exit() {
     set -xe
   fi
 
+  # update project nextflow history
+  update_project_history
+
+  # backup cache
+  if [[ $no_future_resume == false ]]; then
+    echo "=== Execution complete — uploading Nextflow cache and history file"
+    # only upload cache.tar(cache and history)
+    # files in workdir $DX_PROJECT_CONTEXT_ID:/nextflow_cache_db/$NXF_UUID/work/ are uploaded by the plugin after each subjob
+
+    # wrap cache folder and history and upload cache.tar
+    if [[ -n "$(ls -A .nextflow)" ]]; then
+      tar -cf cache.tar .nextflow
+      # remove any existing cache.tar with the same session id
+      dx rm "$DX_PROJECT_CONTEXT_ID:/nextflow_cache_db/$NXF_UUID/cache.tar" 2>&1 >/dev/null || true
+
+      CACHE_ID=$(dx upload "cache.tar" --path "$DX_PROJECT_CONTEXT_ID:/nextflow_cache_db/$NXF_UUID/cache.tar" --no-progress --brief --wait -p -r) &&
+        echo "Upload cache of current session as file: $CACHE_ID" &&
+        rm -f cache.tar ||
+        echo "Failed to upload cache of current session $NXF_UUID"
+    else
+      echo "No cache is generated from this execution. Skip uploading cache."
+    fi
+
+  # no_future_resume is true
+  # clean up files of this session
+  else
+    echo "=== Execution complete — removing working files from workdir"
+    dx tag "$DX_JOB_ID" "no_future_resume"
+    nextflow clean -f $NXF_UUID || true
```

Comment:
Here I use the native clean cmd to clean up the actual workdir (which can be specified with `-w` in the run cmd)

But I wonder if we want to do both 1) skipping caching and 2) clean up workdir, or just 1)?
@commandlinegirl